src/components/badge/index.js

```diff
@@ -107,12 +112,43 @@ export default class Badge extends BaseComponent {
       width += borderWidth * 2;
       height += borderWidth * 2;
     }
-
     return {width, height};
   }
 
   renderLabel() {
-    const {label} = this.props;
+    const {labelLengthFormater} = this.props;
+    let {label} = this.props;
```

Comment:
In case the label is not numeric (e.g "99+", "abc") this whole functionality is irrelevant and should be ignored.

Reply:
Done. Added isNaN check